web/src/message_view_header.ts

```diff
@@ -59,6 +71,21 @@ function get_message_view_header_context(filter: Filter | undefined): MessageVie
         title,
         is_spectator: page_params.is_spectator,
     });
+    if (title === "Mentions") {
```

Comment:
You can't check the titles this way, because they ARE tagged for translation -- so it'll be different in other languages. I think you need `filter.get_title` to be doing these computations, and this function just displaying what it passed through.

Take a read through https://zulip.readthedocs.io/en/latest/translating/internationalization.html.